### Note
* Instructions have been included for each segment. You do not have to follow them exactly, but they are included to help you think through the steps.

In [1]:
# Dependencies and Setup
import pandas as pd

# File to Load (Remember to Change These)
file_to_load = "Resources/purchase_data.csv"

# Read Purchasing File and store into Pandas data frame
df_purchase = pd.read_csv(file_to_load)

## Player Count

* Display the total number of players


In [2]:
# Let's take a look at the data
df_purchase.head()

Purchase ID             SN  Age Gender  Item ID  \
0            0        Lisim78   20   Male      108   
1            1    Lisovynya38   40   Male      143   
2            2     Ithergue48   24   Male       92   
3            3  Chamassasya86   24   Male      100   
4            4      Iskosia90   23   Male      131   

                                   Item Name  Price  
0  Extraction, Quickblade Of Trembling Hands   3.53  
1                          Frenzied Scimitar   1.56  
2                               Final Critic   4.88  
3                                Blindscythe   3.27  
4                                       Fury   1.44

In [3]:
# Display the total number of players
df_unique_sn = df_purchase["SN"].drop_duplicates()
players_count = len(df_unique_sn)
players_count

576

## Purchasing Analysis (Total)

* Create a summary data frame to hold the results

* Create a summary data frame to hold the results


* Optional: give the displayed data cleaner formatting


* Display the summary data frame


In [4]:
# Run basic calculations to obtain number of unique items, average price, etc.
df_purchase.describe().T

count        mean         std  min     25%     50%     75%  \
Purchase ID  780.0  389.500000  225.310896  0.0  194.75  389.50  584.25   
Age          780.0   22.714103    6.659444  7.0   20.00   22.00   25.00   
Item ID      780.0   91.755128   52.697702  0.0   47.75   92.00  138.00   
Price        780.0    3.050987    1.169549  1.0    1.98    3.15    4.08   

                max  
Purchase ID  779.00  
Age           45.00  
Item ID      183.00  
Price          4.99

In [5]:
# Getting number of unique items

df_unique_items = df_purchase["Item ID"].drop_duplicates()
items_count = len(df_unique_items)
items_count
df_unique_items

0      108
1      143
2       92
3      100
4      131
      ... 
664     47
673    130
700     90
717    177
727    104
Name: Item ID, Length: 179, dtype: int64

In [6]:
# Getting average price
average_price = df_purchase["Price"].mean()
average_price

3.050987179487176

## Gender Demographics

In [7]:
df_purchase["Gender"].drop_duplicates()

0                      Male
9     Other / Non-Disclosed
15                   Female
Name: Gender, dtype: object

In [8]:
# Percentage and Count of Male Players
df_purchase_male = df_purchase[df_purchase["Gender"] == "Male"]

# Obtaining Male Unique Male Players
df_unique_male_sn = df_purchase_male["SN"].drop_duplicates()
male_player_count = len(df_unique_male_sn)

print("Total Unique Male Players = ", male_player_count)

print("Total Unique Male Players = ", male_player_count/ players_count * 100, "%" )

Total Unique Male Players =  484
Total Unique Male Players =  84.02777777777779 %


In [9]:
# Percentage and Count of Female Players
df_purchase_female = df_purchase[df_purchase["Gender"] == "Female"]
len(df_purchase_female)

# Obtaining Male Unique Female Players
df_unique_female_sn = df_purchase_female["SN"].drop_duplicates()
female_player_count = len(df_unique_female_sn)
female_player_count

print("Total Unique Female Players = ", female_player_count)

print("Total Unique Female Players = ", female_player_count/ players_count * 100, "%" )

Total Unique Female Players =  81
Total Unique Female Players =  14.0625 %


In [10]:
# Percentage and Count of Other / Non_Discloused Players
df_purchase_other = df_purchase[df_purchase["Gender"] == "Other / Non-Disclosed"]
len(df_purchase_other)

# Obtaining Other Unique Players
df_unique_other_sn = df_purchase_other["SN"].drop_duplicates()
other_player_count = len(df_unique_other_sn)
other_player_count

print("Total Other Players = ", other_player_count)

print("Total Other Players = ", other_player_count/ players_count * 100, "%" )

Total Other Players =  11
Total Other Players =  1.9097222222222223 %


## Purchasing Analysis (Gender)

In [11]:
# Purchase Count by Male
len(df_purchase_male)

652

In [12]:
# Average Purchase Price by Male
df_purchase_male["Price"].mean()

3.0178527607361953

In [13]:
# Total Purchase Value by Male
df_purchase_male["Price"].sum()

1967.64

In [14]:
# Average Purchase Total per Person by Male
df_purchase_male[["SN","Price"]].groupby(["SN"]).mean()

Price
SN                    
Adairialis76  2.280000
Aeda94        4.910000
Aela59        4.320000
Aelaria33     1.790000
Aelastirin39  3.645000
...                ...
Yasur85       1.100000
Yathecal72    3.460000
Yathedeu43    3.010000
Zhisrisu83    3.945000
Zontibe81     2.676667

[484 rows x 1 columns]

In [15]:
# Purchase Count by Female
len(df_purchase_female)

113

In [16]:
# Average Purchase Price by Female
df_purchase_female["Price"].mean()

3.203008849557519

In [17]:
# Total Purchase Value by Female
df_purchase_female["Price"].sum()

361.94

In [18]:
# Average Purchase Total per Person by female
df_purchase_female[["SN","Price"]].groupby(["SN"]).mean()

Price
SN                     
Adastirin33    4.480000
Aerithllora36  4.320000
Aethedru70     3.540000
Aidain51       3.450000
Aiduesu86      4.480000
...                 ...
Wailin72       2.730000
Yana46         3.550000
Yarithrin84    4.140000
Yathecal82     2.073333
Yoishirrala98  4.580000

[81 rows x 1 columns]

In [19]:
# Purchase Count by Other
len(df_purchase_female)

113

In [20]:
# Average Purchase Price by Other
df_purchase_other["Price"].mean()

3.3460000000000005

In [21]:
# Total Purchase Value by Other
df_purchase_other["Price"].sum()

50.19

In [22]:
# Average Purchase Total per Person by Other
df_purchase_other[["SN","Price"]].groupby(["SN"]).mean()

Price
SN                 
Airi27        3.450
Chanosian48   3.760
Eodaisu60     3.940
Haerithp41    4.400
Idairin51     3.500
Jiskirran77   3.390
Lirtim36      1.330
Maluncil97    2.640
Rairith81     2.220
Siarithria38  3.455
Sundim98      4.750

## Age Demographics

* Establish bins for ages


* Categorize the existing players using the age bins. Hint: use pd.cut()


* Calculate the numbers and percentages by age group


* Create a summary data frame to hold the results


* Optional: round the percentage column to two decimal points


* Display Age Demographics Table


## Purchasing Analysis (Age)

In [23]:
# Bin the purchase_data data frame by age

bins = [0, 9.9, 14.9, 19.9, 24.9, 29.9, 34.9, 39.9, 44.9]
group_names = ["<10", "10-14", "15-19", "20-24", "25-29", "30-34", "35-39", "+40"]

df_purchase["Age Group"] = pd.cut(df_purchase["Age"], bins, labels= group_names, include_lowest=True)
df_purchase.head()

Purchase ID             SN  Age Gender  Item ID  \
0            0        Lisim78   20   Male      108   
1            1    Lisovynya38   40   Male      143   
2            2     Ithergue48   24   Male       92   
3            3  Chamassasya86   24   Male      100   
4            4      Iskosia90   23   Male      131   

                                   Item Name  Price Age Group  
0  Extraction, Quickblade Of Trembling Hands   3.53     20-24  
1                          Frenzied Scimitar   1.56       +40  
2                               Final Critic   4.88     20-24  
3                                Blindscythe   3.27     20-24  
4                                       Fury   1.44     20-24

In [24]:
# Purchase Count
df_purchase[["Age Group", "Purchase ID"]].groupby(["Age Group"]).count()

Purchase ID
Age Group             
<10                 23
10-14               28
15-19              136
20-24              365
25-29              101
30-34               73
35-39               41
+40                 12

In [25]:
# Average Purchase Price 
df_purchase[["Age Group", "Price"]].groupby(["Age Group"]).mean()

Price
Age Group          
<10        3.353478
10-14      2.956429
15-19      3.035956
20-24      3.052219
25-29      2.900990
30-34      2.931507
35-39      3.601707
+40        3.045000

In [26]:
# Total Purchase Value
df_purchase[["Age Group", "Price"]].groupby(["Age Group"]).sum()

Price
Age Group         
<10          77.13
10-14        82.78
15-19       412.89
20-24      1114.06
25-29       293.00
30-34       214.00
35-39       147.67
+40          36.54

## Top Spenders

In [27]:
# Average Purchase Total per Person by Age Group

df_purchase[["SN", "Price", "Purchase ID"]].groupby(["SN"], as_index=False).agg({
    "Purchase ID": "count", 
    "Price": ["mean","sum"]
}).sort_values(("Price", "sum"), ascending=False).reset_index().head(5)


index           SN Purchase ID     Price       
                           count      mean    sum
0   360    Lisosia93           5  3.792000  18.96
1   246  Idastidru52           4  3.862500  15.45
2   106   Chamjask73           3  4.610000  13.83
3   275       Iral74           4  3.405000  13.62
4   281  Iskadarya95           3  4.366667  13.10

## Most Popular Items

In [28]:
# Identify the the top 5 spenders in the game by total purchase value, then list (in a table):

df_purchase[["Item ID", "Price", "Purchase ID"]].groupby(["Item ID"], as_index=False).agg({
    "Purchase ID": "count", 
    "Price": ["mean","sum"]
}).sort_values(("Purchase ID", "count"), ascending=False).reset_index().head(5)


index Item ID Purchase ID     Price       
                      count      mean    sum
0    90      92          13  4.614615  59.99
1   174     178          12  4.230000  50.76
2   141     145           9  4.580000  41.22
3   129     132           9  3.221111  28.99
4   105     108           9  3.530000  31.77

## Most Profitable Items

In [29]:

df_purchase[["Item ID", "Price", "Purchase ID"]].groupby(["Item ID"], as_index=False).agg({
    "Purchase ID": "count", 
    "Price": ["mean","sum"]
}).sort_values(("Price", "sum"), ascending=False).reset_index().head(5)


index Item ID Purchase ID     Price       
                      count      mean    sum
0    90      92          13  4.614615  59.99
1   174     178          12  4.230000  50.76
2    80      82           9  4.900000  44.10
3   141     145           9  4.580000  41.22
4   100     103           8  4.350000  34.80

**Observations**

- We can observe in this Data that 84% of Players are Identified as Male.
- The Players group from 20-24 years old buy products more frecuently.
- The Players group from 20-24 years old are producing more revenues.
